lib/_http_client.js

```diff
@@ -562,7 +562,13 @@ function tickOnSocket(req, socket) {
   socket.on('close', socketCloseListener);
 
   if (req.timeout) {
-    socket.once('timeout', () => req.emit('timeout'));
+    const emitRequestTimeout = () => req.emit('timeout');
```

Comment:
I thought along the same lines when I found this bug. The code implementing this feature didn't look like I expected, but I just wanted to make a small, conceptually simple fix to get it merged as quickly as possible.

It think your suggestion would work. What does @evanlucas say?